sycl/test-e2e/CompositeDevice/run_on_composite_device.cpp

```diff
@@ -0,0 +1,58 @@
+// RUN: %{build} -o %t.out
+// RUN: env ZE_FLAT_DEVICE_HIERARCHY=COMBINED %{run} %t.out
+// REQUIRES: level_zero
+
+#include <sycl/sycl.hpp>
+
+#ifdef SYCL_EXT_ONEAPI_COMPOSITE_DEVICE
+
+using namespace sycl::ext::oneapi::experimental;
+
+bool isL0Backend(sycl::backend backend) {
+  return (backend == sycl::backend::ext_oneapi_level_zero);
+}
+
+bool isCombinedMode() {
+  const char *Mode = std::getenv("ZE_FLAT_DEVICE_HIERARCHY");
+  return (Mode != nullptr) && (std::strcmp(Mode, "COMBINED") == 0);
+}
+
+int main() {
+  bool IsCombined = isCombinedMode();
+  auto Platforms = sycl::platform::get_platforms();
+
+  {
+    std::vector<sycl::device> CompositeDevs = get_composite_devices();
+    for (const auto &Composite : CompositeDevs) {
+      auto Backend = Composite.get_backend();
+      auto IsL0 = isL0Backend(Backend);
+      // This test requires L0, and it runs with COMBINED mode, check these
+      // assumptions.
+      assert(IsL0 && IsCombined);
```

Comment:
This isn't really necessary.  We might decide to support composite devices on other backends in the future.  If that happens, you will need to change the test to remove this check.  It would be safe to remove the check now.  If the backend does not support composite devices the call to `get_composite_devices` won't return devices from that backend.

Reply:
Done in https://github.com/intel/llvm/pull/12178/commits/4e78c9067c59b72ccf2f8757e0d0340ad33d473a.